modules/client/src/main/java/org/apache/ignite/internal/client/table/AbstractClientView.java

```diff
@@ -50,6 +62,36 @@ abstract class AbstractClientView<T> implements CriteriaQuerySource<T> {
         this.tbl = tbl;
     }
 
+    /**
+     * Get index mapping from result set to schema.
+     *
+     * @param columns Columns to map.
+     * @param metadata Metadata for query results.
+     * @param startInclusive The first index to cover.
+     * @param endExclusive Index immediately past the last index to cover.
+     * @return Index mapping.
+     */
+    protected static int[] indexMapping(
+            ClientColumn[] columns,
+            int startInclusive,
+            int endExclusive,
+            ResultSetMetadata metadata
+    ) {
+        return Arrays.stream(columns, startInclusive, endExclusive)
```

Comment:
Stream usage seems wasteful here (extra allocations), especially since we know the size of the resulting array.

Reply:
Replaced with array usage